cpp/extensions/cloud_optics/mo_cloud_optics.h

```diff
@@ -62,20 +62,7 @@ class CloudOptics : public OpticalProps {
   real1d pade_sizreg_ssaice; // (nbound)
   real1d pade_sizreg_asyice; // (nbound)
 
-
-
-  CloudOptics() {
-    int icergh = 0;
-    real radliq_lwr = 0;
-    real radliq_upr = 0;
-    real radice_lwr = 0;
-    real radice_upr = 0;
-    int liq_nsteps = 0;
-    int ice_nsteps = 0;
-    real liq_step_size = 0;
-    real ice_step_size = 0;
-  }
-
+  CloudOptics() = default;
```

Comment:
Keep in mind that default constructor does _not_ zero-initialize. If zero-initialization is important, you need to revert this change or add `= 0;` next to the member declaration.

Reply:
Couple things. First, notice these are all local variable decls, not members. I think the coder intended to change members, since otherwise it makes no sense to do this, but my change doesn't actually change how the code behaves. Second, are you sure this will not zero initialize the members? Doesn't the default constructor promise to call `()` on all members, which for primitives, would zero initialize?

Reply:
Also, my valgrind runs were not reporting uninitialized memory use.

Reply:
Oh, you're right. I didn't read carefully the code. Then yes, what you did is perfect.

Reply:
I was likely the original coder, and I agree, that constructor is useless and certainly a mistake on my part. Thanks for correcting that.